docs/extend/admin.md

```diff
@@ -1 +1,159 @@
+  type: 'select',
+  options: {
+    'LOH': 'Liquid Fuel', // The key in this array is what the setting will be stored as in the database, the value is the label the admin will see (remember to use translations if they make sense in your context).
+    'RDX': 'Solid Fuel',
+  },
+  default: 'LOH',
+}
+```
+
+
+If you want to add something to the settings like some extra text or a more complicated input, you can also pass a callback as the first argument that returns JSX. This callback will be executed in the context of [`ExtensionPage`](https://api.docs.flarum.org/js/master/class/src/admin/components/extensionpage.js~extensionpage) and setting values will not be automatically serialized.
+
+### Registering Permissions
+
+New in beta 15, permissions can now be found in 2 places. Now, you can view each extension's individual permissions on their page. All permissions can still be found on the permissions page.
+
+In order for that to happen, permissions must be registered with `ExtensionData`. This is done in a similar way to settings, call `registerPermission`. 
+
+Arguments: 
+ * Permission object
+ * What type of permission - see [`PermissionGrid`](https://api.docs.flarum.org/js/master/class/src/admin/components/permissiongrid.js~permissiongrid)'s functions for types (remove items from the name)
+ * `ItemList` priority
+ 
+Back to our favorite rocket extension:
+
+```js
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+    .registerPermission(
+      {
+        icon: 'fas fa-rocket', // Font-Awesome Icon
+        label: app.translator.trans('acme-interstellar.admin.permissions.fly_rockets_label'), // Permission Label
+        permission: 'discussion.rocket_fly', // Actual permission name stored in database (and used when checking permission).
+      }, 
+      'start', // Category permission will be added to on the grid
+      95 // Optional: Priority
+    );
+});
+```
+
+### Chaining Reminder
+
+Remember these functions call all be chained like:
+
+```js
+app.extensionData
+    .for('acme-interstellar')
+    .registerSetting()
+    .registerSetting()
+    .registerPermission()
+    .registerPermission();
+```
+
+### Extending/Overriding the Default Page
+
+Sometimes you have more complicated settings that mess with relationships, or just want the page to look completely different. In this case, you will need to tell `ExtensionData` that you want to provide your own page.
+
+Simply run `registerPage`:
+
+```js
+
+import StarPage from './components/StarPage';
+
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+    .registerPage(StarPage);
+});
+```
+
+This page will be shown instead of the default.
+
+You can extend the [`ExtensionPage`](https://api.docs.flarum.org/js/master/class/src/admin/components/extensionpage.js~extensionpage) or extend the base `Page` and design your own!
```

Comment:
Maybe show a basic example of extending ExtensionPage to replace the settings content?